### ISSUE TITLE
Elongated memory data for debug_traceTransaction

### ISSUE THREAD

**Issue by antazoey** [NomicFoundation/hardhat#3858]
### Version of Hardhat

2.13.1

### What happened?

When upgrading from 2.13.0 to 2.13.1 or greater, I get lots of empty memory, and realistic or larger txns, I get this bug: https://github.com/NomicFoundation/hardhat/issues/3272

I am using HTTP to send txns to a Hardhat node. The `memory` field of the revert struct log is extremely massive and filled with 0-bytes. Just a bunch of 0s all the way down.

I do the same thing with ganache and anvil and the return value and struct memory make a lot more sense.

### Minimal reproduction steps

This example only works on small txns. Large one cause this bug: https://github.com/NomicFoundation/hardhat/issues/3858

* Invoke a transaction to a Hardhat node using HTTP and store the txn hash
* Call `debug_traceTransaction` using HTTP to observe the trace. Look at the last frame. It looks like this:

```
{
    'pc': 93, 
    'op': 'REVERT', 
    'gas': 29976633, 
    'gasCost': 0, 
    'depth': 1, 
    'stack': [
        '0x000000000000000000000000000000000000000000000000000000003ccfd60b', 
        '0x0000000000000000000000000000000000000000000000000000000000000033', 
        '0x0000000000000000000000000000000000000000000000000000000000000004', 
        '0x0000000000000000000000000000000000000000000000000000000000000080'], 
    'memory': [
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000080', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x82b4290000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000',  
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000',
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000', 
        '0x0000000000000000000000000000000000000000000000000000000000000000'
    ],
    'storage': {
        HexBytes('0x0000000000000000000000000000000000000000000000000000000000000000'): '0x000000000000000000000000318b469bba396aec2c60342f9441be36a1945174'
    }
}
```

Notice the pointless large amounts of empty zeroes it is sending me.

Comparing with `geth`, `ganache`, `eth-tester`, and `anvil`, they dont do this

### Search terms

debug_traceTransaction memory

**Comment by antazoey** [comment]
tldr; when upgrading from 2.13 to any version getter, I get this bug: https://github.com/NomicFoundation/hardhat/issues/3272

**Comment by fvictorio** [comment]
Thanks @antazoey, I'll look into this as soon as I can. The memory array in the `debug_tt` entries has always been a pain in the ass :disappointed:

**Comment by fvictorio** [comment]
Minimal reproduction:

```js
  // push0 push0 mstore push0
  const bytecode = "0x5F5F525F"

  const [account] = await network.provider.send("eth_accounts", [])

  const address = "0x1234567890123456789012345678901234567890"

  await network.provider.send("hardhat_setCode", [address, bytecode])

  const tx = await network.provider.send("eth_sendTransaction", [{
    from: account,
    to: address
  }])

  const trace = await network.provider.send("debug_traceTransaction", [tx])

  console.log(JSON.stringify(trace, null, 2))
```

That bytecode pushes two 0s to the stack and executes a `MSTORE`, and the resulting memory entry, instead of having a length of 1, has a length of 256. That's a big clue of what can be going on here.
